### PR TITLE
Avoid the old &vec[0] idiom when possible.

### PR DESCRIPTION
**Pull request by drwells**
It is cleaner to just let the array decay into a pointer or use `Container::data()` when appropriate.

### CODE REVIEW

**Comment by masterleinad** [comment]
Hmm... I don't consider the array to pointer decay a particularly good feature and would prefer something that is agnostic to the type of container used.

**Comment by drwells** [comment]
I think the only option for those (that is not pointer decay) is to use `std::begin` or `std::cbegin`. That looks weird enough that I prefer `&v[0]` for fixed-size C arrays, so lets just leave that out for now (its cosmetic in this case anyway).

File: source/numerics/kdtree.cc [hunk 1]
```diff
@@ -67,7 +67,7 @@ KDTree<dim>::get_closest_points(const Point<dim> & target,
   std::vector<unsigned int> indices(n_points);
   std::vector<double>       distances(n_points);
 
-  kdtree->knnSearch(&target[0], n_points, &indices[0], &distances[0]);
+  kdtree->knnSearch(&target[0], n_points, indices.data(), distances.data());
```
Comment (masterleinad) [inline source/numerics/kdtree.cc]: You could probably use
```suggestion
  kdtree->knnSearch(target.begin_raw(), n_points, indices.data(), distances.data());
```
or did you avoid accessing raw memory here purposefully?

Comment (masterleinad) [reply source/numerics/kdtree.cc]: In line 51 as well

**Pull request merged**